pkgs/tools/security/rnp/default.nix

```diff
@@ -0,0 +1,57 @@
+{ lib
+, stdenv
+, botan2
+, bzip2
+, cmake
+, fetchFromGitHub
+, fetchurl
+, json_c
+, zlib
+}:
+let
+  versionCmake = fetchurl {
+    url = "https://raw.githubusercontent.com/riboseinc/cmake-versioning/c78a0be/version.cmake";
+    sha256 = "09ghbxyf6kr75857a3v9zql5j1zxnw1fkbndwsyrc1k9qmg24824";
+  };
+  rev = "ffcfb6319ffd5dfe0146bc119a3c8eb994c132a2";
+in
+stdenv.mkDerivation rec {
+  pname = "rnp";
+  version = "v0.14.0-1-g${builtins.substring 0 7 rev}+${toString SOURCE_DATE_EPOCH}";
+
+  # For generating the correct timestamp in cmake
+  SOURCE_DATE_EPOCH = 1613923200;
```

Comment:
Time stamps are usually not corrected in nixos. Why is this necessary here? I would prefer to just keep it at its default.

Reply:
Thanks for the feedback.

I'd like to know if there's a way to have two kinds of "version" attributes, one for Nix (to keep to nixpkgs's standards) and one for the build process to use (to make it not fail).  If not, then I'll have to come up with something else.

The reason I ask is that since this package is built from a Git commit, according to Nix's packaging guidelines the version should contain `unstable` and the date in `yyyy-mm-dd` format.  However, setting this would break the build process --- it parses its full name and tries to match it with a specific regex pattern, which is unfulfilled by e.g. `unstable-2021-02-21`, but is fulfilled by the string in the above snippet.

Your help is appreciated.

Reply:
You can just patch the right version number in postPatch and give it some dummy version to be happy or widen the regex.